### PR TITLE
26746 - Avoid translating several times post type and taxonomy default labels

### PR DESCRIPTION
**Pull request by audrasjb**
Trac ticket: https://core.trac.wordpress.org/ticket/26746

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/default-filters.php [hunk 1]
```diff
@@ -602,7 +602,7 @@
 
 // Taxonomy.
 add_action( 'init', 'create_initial_taxonomies', 0 ); // Highest priority.
-add_action( 'change_locale', 'create_initial_taxonomies' );
+add_action( 'change_locale', 'create_initial_taxonomies', 0 );
```
Comment (peterwilsoncc) [inline src/wp-includes/default-filters.php]: Back-compat break for developers running `remove_action()` at the default priority. Why do these need to change priority zero?

Same for posts too.

File: src/wp-includes/class-wp-post-type.php [hunk 2]
```diff
@@ -44,6 +44,14 @@ final class WP_Post_Type {
 	 */
 	public $labels;
 
+	/**
+	 * Default labels
+	 *
+	 * @since 6.0.0
+	 * @var (string|null)[][]
```
Comment (costdev) [inline src/wp-includes/class-wp-post-type.php]: Missing var name.
```
* @var (string|null)[][] $default_labels
```

File: src/wp-includes/class-wp-post-type.php [hunk 3]
```diff
@@ -785,4 +793,69 @@ public function get_rest_controller() {
 
 		return $this->rest_controller;
 	}
+
+	/**
+	 * Returns the default labels for post types.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @return (string|null)[][]
```
Comment (costdev) [inline src/wp-includes/class-wp-post-type.php]: Missing return description.
```
* @return (string|null)[][] The default labels for post types.
```

File: src/wp-includes/class-wp-taxonomy.php [hunk 4]
```diff
@@ -42,6 +42,14 @@ final class WP_Taxonomy {
 	 */
 	public $labels;
 
+	/**
+	 * Default labels
+	 *
+	 * @since 6.0.0
+	 * @var (string|null)[][]
```
Comment (costdev) [inline src/wp-includes/class-wp-taxonomy.php]: Missing var name.
```
 * @var (string|null)[][] $default_labels
```

File: src/wp-includes/class-wp-taxonomy.php [hunk 5]
```diff
@@ -562,4 +570,71 @@ public function get_rest_controller() {
 
 		return $this->rest_controller;
 	}
+
+	/**
+	 * Returns the default labels for taxonomies.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @return (string|null)[][]
```
Comment (costdev) [inline src/wp-includes/class-wp-taxonomy.php]: Missing return description.
```
* @return (string|null)[][] The default labels for taxonomies.
```

File: src/wp-includes/class-wp-post-type.php [hunk 6]
```diff
@@ -44,6 +44,14 @@ final class WP_Post_Type {
 	 */
 	public $labels;
 
+	/**
+	 * Default labels
```
Comment (costdev) [inline src/wp-includes/class-wp-post-type.php]:
```suggestion
	 * Default labels.
```

File: src/wp-includes/class-wp-taxonomy.php [hunk 7]
```diff
@@ -42,6 +42,14 @@ final class WP_Taxonomy {
 	 */
 	public $labels;
 
+	/**
+	 * Default labels
```
Comment (costdev) [inline src/wp-includes/class-wp-taxonomy.php]:
```suggestion
	 * Default labels.
```

**Review by costdev — COMMENTED** [review]
Some docblock suggestions

File: src/wp-includes/class-wp-post-type.php (hunk 2 above)
Comment (audrasjb) [reply src/wp-includes/class-wp-post-type.php]: Thx!

File: src/wp-includes/default-filters.php (hunk 1 above)
Comment (audrasjb) [reply src/wp-includes/default-filters.php]: Indeed I removed the priority override and it still works as expected.

**Comment by audrasjb** [comment]
Thanks @costdev @peterwilsoncc for the review!

**Review by peterwilsoncc — APPROVED** [review]
Looks good to me.

**Comment by peterwilsoncc** [comment]
Merged in https://core.trac.wordpress.org/changeset/53058

**Pull request closed without merging**